core/trino-main/src/main/java/io/trino/operator/output/PagePartitionerPool.java

```diff
@@ -74,7 +76,18 @@ public void release(PagePartitioner pagePartitioner)
     public void close()
     {
         // pagePartitioner.close can take a long time (flush->serialization), we want to keep it out of the synchronized block
-        markClosed().forEach(PagePartitioner::close);
+        Collection<PagePartitioner> toClose = markClosed();
+        closeSafely(toClose);
+    }
+
+    private static void closeSafely(Collection<PagePartitioner> toClose)
+    {
+        try (Closer closer = Closer.create()) {
+            toClose.forEach(closer::register);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
you should release references to `PagePartitioners` in `finally`

Reply:
Instead, I added clear to the `markClosed`